tests/model_serving/model_server/multi_node/conftest.py

```diff
@@ -96,3 +123,84 @@ def multi_node_predictor_pods_scope_class(
         client=admin_client,
         isvc=multi_node_inference_service,
     )
+
+
+@pytest.fixture(scope="function")
+def patched_multi_node_isvc_external_route(
+    multi_node_inference_service: InferenceService,
+) -> Generator[InferenceService, Any, Any]:
+    with ResourceEditor(
+        patches={
+            multi_node_inference_service: {
+                "metadata": {"labels": {Labels.Kserve.NETWORKING_KSERVE_IO: Labels.Kserve.EXPOSED}},
+            }
+        }
+    ):
+        for sample in TimeoutSampler(
+            wait_timeout=Timeout.TIMEOUT_1MIN,
+            sleep=1,
+            func=lambda: multi_node_inference_service.instance.status,
+        ):
+            if sample and sample.get("url", "").startswith(Protocols.HTTPS):
+                break
+
+        yield multi_node_inference_service
+
+
+@pytest.fixture(scope="function")
+def patched_multi_node_worker_spec(
+    request: FixtureRequest,
+    multi_node_inference_service: InferenceService,
+) -> Generator[InferenceService, Any, Any]:
+    with ResourceEditor(
+        patches={
+            multi_node_inference_service: {
+                "spec": {
+                    "predictor": {"workerSpec": request.param["worker-spec"]},
+                },
+            }
+        }
+    ):
+        yield multi_node_inference_service
+
+
+@pytest.fixture()
+def ray_ca_tls_secret(admin_client: DynamicClient) -> Secret:
+    return Secret(
```

Comment:
Can you use ensure_exists=True here? that way, you don't have to make the api call in test.

Reply:
no; this is the test.